bigframes/session/_io/bigquery.py

```diff
@@ -17,15 +17,45 @@
 import datetime
 import textwrap
 import types
-from typing import Dict, Iterable, Union
+from typing import Dict, Iterable, Optional, Sequence, Union
 import uuid
 
 import google.cloud.bigquery as bigquery
 
 IO_ORDERING_ID = "bqdf_row_nums"
+MAX_LABELS_COUNT = 64
 TEMP_TABLE_PREFIX = "bqdf{date}_{random_id}"
 
 
+def create_job_configs_labels(
+    job_configs_labels: Optional[Dict[str, str]],
+    api_methods: Sequence[str],
+) -> Dict[str, str]:
+    # If there is no label set
+    if job_configs_labels is None:
+        labels = {}
+        label_values = list(api_methods)
+    else:
+        labels = job_configs_labels.copy()
+        cur_labels_len = len(job_configs_labels)
+        api_methods_len = len(api_methods)
+        # If the total number of labels is under the limit of labels count
+        if cur_labels_len + api_methods_len <= MAX_LABELS_COUNT:
+            label_values = list(api_methods)
+        # We capture the latest label if it is out of the length limit of labels count
+        else:
+            added_api_len = cur_labels_len + api_methods_len - MAX_LABELS_COUNT
+            label_values = list(api_methods)[-added_api_len:]
+
+    for i, label_value in enumerate(label_values):
+        if job_configs_labels is not None:
+            label_key = "bigframes-api-" + str(i + len(job_configs_labels))
+        else:
+            label_key = "bigframes-api-" + str(i)
+        labels[label_key] = label_value
+    return labels
```

Comment:
This all seems way more complicated than it needs to be. Some functional-style code could make this a lot more understandable. Note: You'll need to import `itertools` for `chain`, which is a built-in module.

```suggestion
    # If there is no label set
    if job_configs_labels is None:
        job_configs_labels = {}

    labels = list(itertools.chain(
        job_config_labels.keys(),
        (f"bigframes-api-{i}" for i in range(len(api_methods)))
    ))
    values = list(itertools.chain(job_config_labels.values(), api_methods))
    return dict(zip(labels[:MAX_LABELS_COUNT], values[:MAX_LABELS_COUNT]))
```